packages/website/src/components/linter/bridge.ts

```diff
@@ -0,0 +1,95 @@
+import { createSystem } from '@typescript/vfs';
+import type * as Monaco from 'monaco-editor';
+import type * as ts from 'typescript';
+
+import type { ConfigModel, PlaygroundSystem } from '../playground/types';
+import { debounce } from '../util/debounce';
+
+export async function addLibFiles(
+  system: PlaygroundSystem,
+  monaco: typeof Monaco,
+): Promise<void> {
+  const worker = await monaco.languages.typescript.getTypeScriptWorker();
+  const workerInstance = await worker();
+  const libs = await workerInstance.getLibFiles();
+  if (libs) {
+    for (const [name, content] of Object.entries(libs)) {
+      system.writeFile('/' + name, content);
+    }
+  }
+}
+
+export function createFileSystem(config: ConfigModel): PlaygroundSystem {
+  const files = new Map<string, string>();
+  files.set(`/.eslintrc`, config.eslintrc);
+  files.set(`/tsconfig.json`, config.tsconfig);
+  files.set(`/file.${config.fileType}`, config.code);
+
+  const fileWatcherCallbacks = new Map<string, Set<ts.FileWatcherCallback>>();
+  const directoryWatcherCallback = new Set<ts.DirectoryWatcherCallback>();
+
+  const system = createSystem(files) as PlaygroundSystem;
+
+  system.watchFile = (path, callback, pollingInterval): ts.FileWatcher => {
```

Comment:
Is setting `watchFile` or other properties on a TS system documented anywhere? I'm poking through https://github.com/microsoft/TypeScript-Website/tree/v2/packages/typescript-vfs and don't see anything.

Reply:
they are part of `ts.System`, they are not required to be implemented and mostly used by watch program, 

![image](https://user-images.githubusercontent.com/625469/226489633-dd114c87-9d30-45f3-b6b7-e639f60ef6d5.png)

Reply:
I guess I was just hoping for first party docs explaining this - @jakebailey's been looking at documenting more of the TS APIs.